lib/Sema/CSApply.cpp

```diff
@@ -8993,10 +8993,16 @@ ExprWalker::rewriteTarget(SolutionApplicationTarget target) {
     result.setExpr(resultExpr);
 
     if (cs.isDebugMode()) {
+      // If target is a multi-statement closure or
+      // a tap expression, expression will not be fully
+      // type checked until these types are visited in
+      // processDelayed().
       auto &log = llvm::errs();
-      log << "---Type-checked expression---\n";
-      resultExpr->dump(log);
-      log << "\n";
+      if (!ClosuresToTypeCheck.empty() || !TapsToTypeCheck.empty()) {
+        log << "---Partially type-checked expression---\n";
+        resultExpr->dump(log);
+        log << "\n";
+      }
```

Comment:
I think we want to print in `else` case as well otherwise we’d loose output for closure body elements if there are multiple closures or interpolations in a given expression.